test/math/QuaternionTest.ooc

```diff
@@ -0,0 +1,88 @@
+		})
+		// Quaternion tests
+        this add("addition", func() {
+            expect(this quaternion0 + this quaternion1 == this quaternion2)
+        })
+        this add("subtraction", func() {
+            expect(this quaternion0 - this quaternion0 == Quaternion new())
+        })
+        this add("multiplication", func() {
+            expect(this quaternion0 * this quaternion1 == this quaternion3)
+        })
+        this add("scalarMultiplication", func() {
+            expect((-1.0f) * this quaternion0 == -this quaternion0)
+        })
+        this add("norm", func() {
+            expect(this quaternion0 norm == 65.5991592f)
+        })
+        this add("ActionOnVector", func() {
+            direction := FloatPoint3D new(1.0f, 1.0f, 1.0f)
+			quaternion := Quaternion createRotation(this toRadians(120.0f), direction) 
+            point1 := FloatPoint3D new(5.0f, 6.0f, 7.0f)
+            point2 := FloatPoint3D new(7.0f, 5.0f, 6.0f)
+            expect((quaternion * point1) distance(point2) == 0.0f)
+        })
+        this add("LogarithmExponential", func() {
+            roll := this toRadians(20.0f)
+            pitch := this toRadians(-30.0f)
+            yaw := this toRadians(45.0f)
+            quaternion := Quaternion createRotationZ(yaw) * Quaternion createRotationY(pitch) * Quaternion createRotationX(roll)
+            expLog := quaternion calculateExponential() calculateLogarithm()
+            logExp := quaternion calculateLogarithm() calculateExponential()
+            expect(expLog real == quaternion real)
+            expect(logExp real == quaternion real)
+            expect(this isWithinTolerance(0, logExp imaginary distance(quaternion imaginary), tolerance))
+            expect(this isWithinTolerance(0, expLog imaginary distance(quaternion imaginary), tolerance))
+        })
+		this add("Action", func() {
+			roll := this toRadians(30.0f)
+			pitch := this toRadians(20.0f)
+			yaw := this toRadians(-45.0f)
+			quaternion := Quaternion createRotationZ(yaw) * Quaternion createRotationY(pitch) * Quaternion createRotationX(roll)
+			expect(this isWithinTolerance(roll, quaternion rotationX, tolerance))
+			expect(this isWithinTolerance(pitch, quaternion rotationY, tolerance))
+			expect(this isWithinTolerance(yaw, quaternion rotationZ, tolerance))	
+		})
+		this add("RotationDirectionRepresentation1", func() {
+			angle := this toRadians(30.0f)
+			direction := FloatPoint3D new(1.0f, 4.0f, 7.0f)
+			direction /= direction norm
+			quaternion := Quaternion createRotation(angle, direction)
+			expect(this isWithinTolerance(angle, quaternion rotation, tolerance))
+		})
+    }
+	//
+	// Helpers
+	//
+	isWithinTolerance: func (target, actualValue, tolerance: Float) -> Bool {
+		actualValue >= target - tolerance abs() && actualValue <= target + tolerance abs()
+	}
+    toRadians: func (angle: Float) -> Float {
+		Float pi / 180.0f * angle
+	}
+	toDegrees: func (angle: Float) -> Float {
+		180.0f / Float pi * angle
+	}
```

Comment:
`toRadians` and `toDegrees` should be added in `source/sdk/math.ooc` if they don't already exist.